### PR TITLE
[do not review]Expose resource resolvers for choice-set Images.

### PR DESCRIPTION
**Pull request by an888ha**
1.Choice Set renderer will ask acrChoiceSetResourceDelegate for choice set images in different states.
2. Tint Color for toggle exposed so that anyone can set tint color , if he/she over-rides the toggle input renderer.

### CODE REVIEW

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRToggleInputDataSource.mm [hunk 1]
```diff
@@ -25,13 +25,17 @@ @implementation ACRToggleInputDataSource
 
 - (instancetype)initWithInputToggle:(std::shared_ptr<ToggleInput> const&)toggleInput
       WithHostConfig:(std::shared_ptr<HostConfig> const&)hostConfig
+       withTintColor:(UIColor*) color;
```
Comment (nesalang) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRToggleInputDataSource.mm]: I suggest to add getter for _toggleSwitch then modifying the interface. that way developers will have more control over what they can customize.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRChoiceSetResourceDelegate.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+//
+//  Created by Ankit Maurya on 8/22/18.
+//  Copyright © 2018 Microsoft. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "ACOAdaptiveCard.h"
+#import "ACOBaseActionElement.h"
+
+@protocol ACRChoiceSetResourceDelegate <NSObject>
```
Comment (nesalang) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRChoiceSetResourceDelegate.h]: I suggest to use ACOResourceResolvers
1. create ACOResourceResolvers, and register custom ACOResourceResolver with scheme.
Scheme can be any NSString
 [customResolvers setResourceResolver:customRsolver scheme: @"ACRChoiseSetResorceResolverSchme"];

2. implement ACOIResourceResolver interface;

3. to retrieve image, add code that query the resourceResolvers with the scheme, and get image using the interface.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRChoiceSetViewDataSource.mm [hunk 3]
```diff
@@ -247,4 +247,49 @@ - (NSString *)getTitlesOfChoices
     return [values componentsJoinedByString:@", "];
 }
 
+- (UIImage*) getRadioButtonForSelectedState{
```
Comment (nesalang) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRChoiceSetViewDataSource.mm]: Please pull from origin master as logic of data source has changed. Images loading happens only in one place. A new UITableViewCell does the loading, and registered with the table for efficient recycling.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRChoiceSetResourceDelegate.h [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+- (UIImage*) getRadioButtonForSelectedState;
```
[the remark this answers is not in the document]
Comment (an888ha) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRChoiceSetResourceDelegate.h]: Ok. I will add an Enum which can handle different states, like selected, unselected, unselectedFouces, selectedFocus.

**Comment by khouzam** [comment]
Closing as it has been superseded by a new PR.

**Pull request closed without merging**